### ISSUE TITLE
aiplatform_v1.ModelGardenServiceClient gets stuck on first initialization when done concurrently

### ISSUE THREAD

**Issue by tmyjoe** [googleapis/python-aiplatform#2805]
When initializing the `aiplatform_v1.ModelGardenServiceClient` class concurrently, we've encountered an issue where it gets stuck.

Upon further investigation, it stops at the following line where a GRPC channel is created in `ModelGardenServiceGrpcTransport`,

https://github.com/googleapis/python-aiplatform/blob/1704db8000bce2732bf0219a2f7da3e2b9379048/google/cloud/aiplatform_v1/services/model_garden_service/transports/grpc.py#L166

Following this further leads to where the `cygrpc.Channe`l is created, where it appears that both threads stop.
https://github.com/grpc/grpc/blob/master/src/python/grpcio/grpc/_channel.py#L1973

Once initialization is successful for the first time without this issue, subsequent processes proceed without any problems even when done concurrently, suggesting that this issue only occurs on the first attempt.

#### Environment details

  - OS type and version: macOS 12.6.6
  - Python version: 3.10.9
  - pip version: 23.2.1
  - `google-cloud-aiplatform` version: `1.35.0`

#### Steps to reproduce

  1. `aiplatform_v1.ModelGardenServiceClient` concurrently

#### Code example

here is the code to reproduce issue.

```python
# example

from google.cloud import aiplatform_v1
from concurrent.futures import ThreadPoolExecutor

def init_model_gardern_service_client(_):
    # Create a client
    client = aiplatform_v1.ModelGardenServiceClient()

def parallel_execution():
    with ThreadPoolExecutor() as executor:
        results = list(executor.map(init_model_gardern_service_client, [None]*2))

if __name__ == "__main__":
    parallel_execution()
```

#### Stack trace

There is no stacktrace as the process gets just stuck.

**Comment by parthea** [comment]
Hi @tmyjoe ,

I can re-create the issue with `grpcio==1.59.0` but not with older versions such as `grpcio==1.54.0`, `grpcio==1.56.0`, `grpcio==1.58.0`. There is a known issue in https://github.com/grpc/grpc/issues/31885 with similar symptoms. I'll add a comment on that bug to say that we're seeing strange behaviour from `grpcio==1.59.0` which was released i the last 2 weeks, compared to earlier versions.

In the meantime, please can you share the specific version of `grpcio` that you're using by running this code in your python environment?

```
import grpc
print(grpc.__version__)
```

**Comment by parthea** [comment]
I'm going to close this issue as a duplicate of https://github.com/grpc/grpc/issues/34672 which has been triaged by the grpc team.

**Comment by tmyjoe** [comment]
Thank you for looking into this issue. I'll try it with the older versions.

**Comment by tmyjoe** [comment]
I've confirmed that the issue has been resolved when I used it with `grpcio==1.58.0`. Thank you.
